lgc/state/PipelineState.cpp

```diff
@@ -985,8 +985,15 @@ bool PipelineState::isTessOffChip() {
 bool PipelineState::isPackInOut() {
   // Pack input/output requirements:
   // 1) -pack-in-out option is on
-  // 2) It is a VS-FS pipeline
-  return PackInOut && (m_stageMask == (shaderStageToMask(ShaderStageVertex) | shaderStageToMask(ShaderStageFragment)));
+  // 2) It supports VS-FS, VS-TCS and TES-FS
+  if (!PackInOut)
+    return false;
+
+  if (hasShaderStage(ShaderStageVertex)) {
+    const unsigned nextStage = getNextShaderStage(ShaderStageVertex);
+    return nextStage == ShaderStageFragment || nextStage == ShaderStageTessControl;
+  }
```

Comment:
This returns true for a `VS-TCS-TES-GS-FS` pipeline, which we don't want -- right?

Reply:
Although packing for GS is not supported yet, packing between VS and TCS can be allowed in this change. (@linqun suggested to relax the requirements.) I have changed the commit title to align with it.

Reply:
My concern would be that that the `isPackInOut` check does not distinguish between stages, so this would enable packing both for the VS-TCS interface and for the GS-FS interface, but the latter isn't actually implemented yet.

Reply:
Ok, I will change it back to make isPackInOut clear.